js/effects.js

```diff
@@ -0,0 +1,48 @@
+import noUiSlider from 'nouislider';
+import 'nouislider/dist/nouislider.min.mjs';
+// мапа с логикой для стилей
+import { EFFECT_OPTION_MAP } from './effect-map';
+
+const form = document.querySelector('.img-upload__form'); // сама форма
+const image = form.querySelector('.img-upload__preview img'); // превью картинка
+const sliderFieldset = form.querySelector('.img-upload__effect-level'); // филдсет слайдера
+const customSliderWrapper = sliderFieldset.querySelector('.effect-level__slider'); // пустой див
+const effectInputs = form.querySelectorAll('.effects__radio'); // все инпуты эффектов
+// создаем новый евент изменения
+const CHANGE_EVENT = new Event('change');
+// с помощью библиотеки создаем массив
+const slider = noUiSlider.create(customSliderWrapper, { // запиши сюда
+  ...EFFECT_OPTION_MAP.none.slider, // из оригинала базовый объект
+  connect: 'lower' // опция нижняя
+});
+// слайдер будет закрытым изначально
+sliderFieldset.hidden = true;
+// проходим по всем инпутам
+effectInputs.forEach(
+  (effectInput) => effectInput.addEventListener('change', () => { // ловим изменение на который нажали
+    const effect = form.effect.value; // значение пойманного инпута
+    sliderFieldset.hidden = effect === 'none'; // говорим что выключенный слайдер имеет стили оригинала
+    slider.updateOptions(EFFECT_OPTION_MAP[effect].slider, false); // меняем кофигурацию слайдера
+    // у пойманного эффекта ипользуя мапу записываем слайдур из мапы
+  })
+);
```

Comment:
Нет. Итерироваться по массиву дорого, лучше подписаться на родителя. Это событие всплывает, так что всё ок
```suggestion
const list = form.querySelector('.effects__list');
list.addEventListener('change', () => {
  const effect = form.effect.value; // значение пойманного инпута
  sliderFieldset.hidden = effect === 'none'; // говорим что выключенный слайдер имеет стили оригинала
  slider.updateOptions(EFFECT_OPTION_MAP[effect].slider, false); // меняем кофигурацию слайдера
  // у пойманного эффекта ипользуя мапу записываем слайдур из мапы
});
```